libs/events/src/EventEmitter.ts

```diff
@@ -0,0 +1,36 @@
+import { EventNames, EventPayloads } from './types'
+
+export type Listener<T extends EventNames> = (payload: EventPayloads[T]) => void
```

Comment:
this is an early version, please see 9b48dbbd8ce3d757312cca91d40e6cc2a28717af

this continued to refine as I worked on this. So if you want to get the idea on the end result, just check the last PR.

I just think is nice you get in this PR the idea on the event emitter role